### ISSUE TITLE
Final Project Review --zw287

### ISSUE THREAD

**Issue by RobinWu218** [medhavi27/Hot-Spotify-Tracks#14]
This project aims to create a model that predicts whether a song is(would be) popular or not based on features of a dataset gathered from Spotify API. It builds both a general model and a artist-specific model to evaluate performances of different algorithms.

Three Things I like:
1. The initial data exploration is solid. You have looked at distributions of individual features as well as their correlations. It is also interesting to know that no single feature alone would make a song popular.
2. Good analysis of wether the model is WMD is included in the report. It informs the audience what this model could potentially be used for.
3. Using more than accuracy as evaluation metrics is great. Precision is a good metric to evaluate the quality of the model predictions.

Things to improve:
1. The feature transformation seems a a bit too complicated and could potentially harm the model. For example, using one hot encoder for artist name is not scalable. If the model were to process thousands of artists' works, the number of features would grow so large that this feature completely overshadows other features.
2. The report studies two models. I think it would be better to include other models such as SVMs, neural networks. Two seems a little bit limited.
3. To improve your model, I suggest split the data into train, validation, and a final test set. Then you can apply k-fold cross validation to select good hyper parameters such as lambda in L2 regularization. The test set is only used at the end and not for model selection. In this way, you can getter more robust models and evaluations.

Overall, I think the project is well executed with room to improvement.
